src/main/java/net/imagej/convert/NumberToNumericTypeConverter.java

```diff
@@ -0,0 +1,56 @@
+/*-
+ * #%L
+ * ImageJ2 software for multidimensional image processing and analysis.
+ * %%
+ * Copyright (C) 2009 - 2022 ImageJ2 developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imagej.convert;
+
+import org.scijava.convert.AbstractConverter;
+
+import net.imglib2.type.numeric.NumericType;
+
+/**
+ * Abstract class for Converters that convert a {@link NumericType} to a
+ * {@link Number}.
+ *
+ * @param <I> input type
+ * @param <O> output type
+ * @author Gabriel Selzer
+ */
+public abstract class NumberToNumericTypeConverter<I extends Number, O extends NumericType<O>>
```

Comment:
What do these `I` and `O` bounds actually buy us? Couldn't this be generalized even further to just `I` and `O` without explicit bounds? And if so: what benefit does this abstract class layer actually provide? It's just the implementation of `convert(Object, Class)`, right? But the casts are unsafe. Shouldn't we do runtime type checking on the `src` input to make sure it really is an `I`, in the same way the `NumericTypeToBigIntegerConverter` does but generalized? And if this does have value, shouldn't we push it up into the scijava-common `org.scijava.convert` base layer?